bsmetadata/preprocessing_tools/wikipedia_desc_utils.py

```diff
@@ -45,16 +45,16 @@ def fetch_website_description_from_keyword(self, keyword: str) -> Optional:
         return self.cache[keyword]
 
     def fetch_entity_description_from_keyword(self, keyword: str) -> str:
-        title = string.capwords(keyword)
-        text = self.wiki_dump_db.get_paragraphs(title)[0].text
-        text = re.sub(r"\((?:[^)(]|\([^)(]*\))*\)", "", text)
-        text = nltk.sent_tokenize(text)[0]
-
-        if text is None:
-            text = self.wiki_dump_db.get_paragraphs(self.redirects_map[keyword])[0].text
-            text = nltk.tokenize.sent_tokenize(text)[0]
-
-        if text is None:
-            text = ""
-
+        # try-catch block : incase the description in not found in the first try block, then it will throw KeyError
+        try:
+            title = string.capwords(keyword)
+            text = self.wiki_dump_db.get_paragraphs(title)[0].text
+            text = re.sub(r"\((?:[^)(]|\([^)(]*\))*\)", "", text)
+            text = nltk.sent_tokenize(text)[0]
+        except KeyError:
+            try:
+                text = self.wiki_dump_db.get_paragraphs(self.redirects_map[keyword])[0].text
+                text = nltk.tokenize.sent_tokenize(text)[0]
+            except KeyError:
+                text = ""
```

Comment:
I have a small question, I was under the impression that we had discussed these lines of code in the previous RP, specifically [here](https://github.com/bigscience-workshop/metadata/pull/75#discussion_r761177998).

Could you give me a little more info on why you changed it?

Reply:
Hi @SaulLu, when I tested the previous code that you had suggested, it was failing for certain cases where it was not able to find the description (throwing `KeyError`) using the code on the first `try` block. So, I changed it back so that instead of failing it can continue and try to use the `redirects_map` feature and see if it can still find the description using it. Otherwise, we set the value to `""`.

Reply:
Few suggestions:
1. Have tokenization and removal of special characters logic outside try catch and pulled out commonly for both scenarios
2. Instead of having a nested try catch,  call fetch_description_from title, if it's none, check the redirects map -> if key exists, call fetch_desc_from_title else catch key error and return "" or None.